internal/setup/setup_test.go

```diff
@@ -1933,6 +1933,162 @@ var setupTests = []setupTest{{
 		`,
 	},
 	relerror: `chisel.yaml: more than one default archive: bar, foo`,
+}, {
+	summary: "Define additional archives in v2-archives",
+	input: map[string]string{
+		"chisel.yaml": `
+			format: v1
+			archives:
+				ubuntu:
+					version: 20.04
+					components: [main]
+					suites: [focal]
+					priority: 10
+					public-keys: [test-key]
+			v2-archives:
+				fips:
+					version: 20.04
+					components: [main]
+					suites: [focal]
+					pro: fips
+					priority: 20
+					public-keys: [test-key]
+			public-keys:
+				test-key:
+					id: ` + testKey.ID + `
+					armor: |` + "\n" + testutil.PrefixEachLine(testKey.PubKeyArmor, "\t\t\t\t\t\t") + `
+		`,
+		"slices/mydir/mypkg.yaml": `
+			package: mypkg
+		`,
+	},
+	release: &setup.Release{
+		Archives: map[string]*setup.Archive{
+			"ubuntu": {
+				Name:       "ubuntu",
+				Version:    "20.04",
+				Suites:     []string{"focal"},
+				Components: []string{"main"},
+				Priority:   10,
+				PubKeys:    []*packet.PublicKey{testKey.PubKey},
+			},
+			"fips": {
+				Name:       "fips",
+				Version:    "20.04",
+				Suites:     []string{"focal"},
+				Components: []string{"main"},
+				Pro:        "fips",
+				Priority:   20,
+				PubKeys:    []*packet.PublicKey{testKey.PubKey},
+			},
+		},
+		Packages: map[string]*setup.Package{
+			"mypkg": {
+				Name:   "mypkg",
+				Path:   "slices/mydir/mypkg.yaml",
+				Slices: map[string]*setup.Slice{},
+			},
+		},
+	},
+}, {
+	summary: "Define archives in v2-archives only",
```

Comment:
This is totally needed but I am wondering if we could do even better by doing the same trick we used to do with `v1-public-keys` and, instead of adding one extra test, running the whole table swapping out `archives` for `v2-archives` to check that everything is compatible. Thoughts? Maybe it is not worth it, but I wonder what your opinion is.

Reply:
No, I think that would be too much work with little to no benefit. As tests go, we are checking if archives defined `v2-archives` do make it to the parsed list. And I think that's enough.

Reply:
But we are not checking every combination right? Let's say that `default` validation happens before merging both in the code, how would we know that we are also doing it correctly for `v2-archives`. The work is something like 20 lines per **file** that we test (see TestParseRelease in https://github.com/canonical/chisel/pull/150/files).

Reply:
Do you think we need to explicitly check every combination?

1. We already check if `archives` works with every combination.
2. We are currently checking if `v2-archives` correctly populate `archives`.

Don't they together accomplish the checking for every combination?

Reply:
> Don't they together accomplish the checking for every combination?

That assumes that everything happens after `archives` is populated. What happens if, as I said, we check for default clashes earlier or if we had separate logic for something else. I would just add the logic, it is not a very big effort and we already liked and validated the approach before.

Reply:
Added https://github.com/letFunny/chisel/pull/23/commits/9bc410498b95b7b5da09b768561f0b5f9d5ab46f. 

Do you think there is value in changing `archives` -> `v2-archives` in `slicer_test.go`?

Reply:
Alright, added the changes in `slicer_test.go` as well.